Tasks/VsTest/vstest.ts

```diff
@@ -105,83 +105,90 @@ function invokeVSTest(testResultsDirectory: string): Q.Promise<number> {
         .then(function(overriddenSettingsFile) {
             locateVSVersion()
                 .then(function(vsVersion) {
-                    setRunInParallellIfApplicable(vsVersion);
-                    setupRunSettingsFileForParallel(runInParallel, overriddenSettingsFile)
-                        .then(function(parallelRunSettingsFile) {
-                            var vstestPath = tl.getVariable("VSTest_" + vsVersion.toFixed(1));
-                            if (!vstestPath) {
-                                tl.error(tl.loc('VstestNotFound', vsVersion));
-                                defer.resolve(1);
-                                return defer.promise;
-                            }
-                            var vstestLocation = path.join(vstestPath, "vstest.console.exe");
-                            var vstest = tl.createToolRunner(vstestLocation);
-
-                            testAssemblyFiles.forEach(function(testAssembly) {
-                                var testAssemblyPath = testAssembly;
-                                //To maintain parity with the behaviour when test assembly was filepath, try to expand it relative to build sources directory.
-                                if (sourcesDirectory && !pathExistsAsFile(testAssembly)) {
-                                    var expandedPath = path.join(sourcesDirectory, testAssembly);
-                                    if (pathExistsAsFile(expandedPath)) {
-                                        testAssemblyPath = expandedPath;
-                                    }
+                    setupRunSettingsFileForTestImpact(vsVersion,overriddenSettingsFile)
```

Comment:
if for some reason this fails we shuld continue to run all tests without TIA on. this shuld be a blocker